_docs_use-cases/use-case-consumption-get-shared-versions-of-an-attribute.md

```diff
@@ -0,0 +1,67 @@
+---
+# Start automatic generation
+permalink: use-case-consumption-get-shared-versions-of-an-attribute
+published: true
+title: "Get shared versions of an Attribute"
+type: use-case
+toc: true
+sidebar:
+  - title: "Integrate enmeshed"
+    nav: "docs_integrate"
+properties:
+  - id: RA27
+  - component: Runtime
+  - layer: Consumption
+  - facade: AttributesFacade
+  - function: getSharedVersionsOfAttribute
+  - description:
+  - feature category: Cross-Identity Attribute sharing
+  - tech category: Attributes
+  - status: DONE
+  - documentation status: DONE
+  - comments:
+  - actor: Identity
+  - trigger: REST API
+  - precondition:
+  - result:
+  - priority: n/a
+  - complexity: n/a
+  - size: n/a
+  - created_at:
+  - changed_at:
+  - api_route_regex: GET /api/v2/Attributes/{id}/Versions/Shared
+  - published: default
+  - link: use-case-consumption-get-shared-versions-of-an-attribute
+require:
+required_by:
+api_route_regex: ^GET /api/v2/Attributes/{id}/Versions/Shared$
+# End automatic generation
+---
+
+{{properties.description}}
+
+{% include properties_list.html %}
+
+This use case allows you to retrieve a list of shared [LocalAttributes]({% link _docs_integrate/data-model-overview.md %}#lcoalattribute) for a given source Attribute.
+In case of [IdentityAttributes]({% link _docs_integrate/data-model-overview.md %}#identityattribute) a list comprising of own shared IdentityAttributes for the specified RepositoryAttribute is returned.
+In case of [RelationshipAttributes]({% link _docs_integrate/data-model-overview.md %}#relationshipattribute) the list contains ThirdPartyRelationshipAttributes you re-shared based on the specified RelationshipAttribute.
+
+## Parameters
+
+- The `attributeId` belonging to a LocalAttribute you would like to know all shared versions of
+- Optionally the returned LocalAttributes can be limited to those shared with specific `peers`.
+- `onlyLatestVersions` omits succeeded versions such that only the most recent version you shared per peer is returned. In detail, this means that if the `attributeId` given as input belongs to a LocalAttribute, whose successor you also already have shared with the peer, the shared copy corresponding to the successor will be returned. By default this is set to be `true`.
+
+## On Success
+
+- A list of shared LocalAttribute versions of the source Attribute belonging to the `attributeId` given as input is returned.
+- If `peers` were speficied, the list is limited to the entries shared with those peers.
+- If `onlyLatestVersions` is disabled, all versions will be returned, even if they already have successors.
+
+## On Failure
+
+- No LocalAttributes can be returned if the `attributeId` correlates to an unknown LocalAttribute.
+- No LocalAttributes can be returned if the `attributeId` correlates to a LocalAttribute with a `shareInfo`.
```

Comment:
Is this really prevented by throwing an error? I have only found a check for whether the input is a RepositoryAttribute in the corresponding "Get shared versions of a RepositoryAttribute" use case. However, I have not found the check here that it must be a RepositoryAttribute or an own shared or peer shared RelationshipAttribute for this use case to be called.

Also note that it is currently possible for own shared IdentityAttributes or ThirdPartyRelationshipAttributes to be shared again and therefore the use case could return more than just an empty array in these cases. This will only change with release v5 (added Request validation) so that this can no longer occur.

Reply:
No, we don't have a dedicated check for this. I was already thinking about the upcoming validation. In that case, we will always return an empty list, calling this use case with an own/peer shared IdentityAttribute. For RelationshipAttributes of course this doesn't hold true.

Reply:
Just for my own understanding of "On Failure". I would have thought that the cases that lead to an error being thrown are listed here. If an empty list is returned for certain LocalAttributes, I would rather consider that a success.

Reply:
Yes you are right. I delete this point, as well as the one regarding unknown peers.